code/modules/research/experiments/improve.dm

```diff
@@ -0,0 +1,41 @@
+/datum/experiment_type/improve
+	name = "Improve"
+	hidden = TRUE
+	var/uses = 0
+
+/datum/experiment/improve
+	weight = 800
+	experiment_type = /datum/experiment_type/improve
+
+/datum/experiment/improve/can_perform(obj/machinery/rnd/experimentor/E,obj/item/O)
+	. = ..()
+	var/datum/experiment_type/improve/mode = E.experiments[/datum/experiment_type/improve]
+	if(!mode || mode.uses <= 0)
+		. = FALSE
+
+/datum/experiment/improve/perform(obj/machinery/rnd/experimentor/E,obj/item/O)
+	. = ..()
+	var/success = FALSE
+	if(istype(O, /obj/item/stock_parts/cell))
+		var/obj/item/stock_parts/cell/C = O
+		C.maxcharge *= 2
+		C.charge *= 2
+		C.chargerate *= 2
+		success = TRUE
+	else if(istype(O, /obj/item/stock_parts))
+		var/obj/item/stock_parts/P = O
+		if(P.rating <= 4) //because someone's going to try to get T100+ parts
+			P.rating *= 3
```

Comment:
If I start with a base rating part, does that get me the ability to make an `improved improved [part]`?